pkg/goldpinger/updater.go

```diff
@@ -18,18 +18,50 @@ import (
 	"fmt"
 	"log"
 	"time"
+
+	"github.com/bloomberg/goldpinger/pkg/models"
+
+	"github.com/cespare/xxhash"
+	rendezvous "github.com/stuartnelson3/go-rendezvous"
 )
 
 func StartUpdater() {
-
 	if GoldpingerConfig.RefreshInterval <= 0 {
 		log.Println("Not creating updater, period is 0")
 		return
 	}
+
+	var pingAllPods bool
+	if GoldpingerConfig.PodIP == "" {
+		log.Println("PodIP not set: pinging all pods")
+		pingAllPods = true
+	}
+
+	n := int(GoldpingerConfig.PingNumber)
+	if n == 0 {
+		log.Println("--ping-number set to 0: pinging all pods")
+		pingAllPods = true
+	}
+
 	// start the updater
 	go func() {
 		for {
-			results := PingAllPods(GetAllPods())
+			allPods := GetAllPods()
+			var results models.CheckResults
+			if pingAllPods || n >= len(allPods) {
+				results = PingAllPods(allPods)
+			} else {
+				rzv := rendezvous.New([]string{}, rendezvous.Hasher(xxhash.Sum64String))
```

Comment:
this could be moved to the top of the closure, with old nodes being removed and new nodes being added, to improve performance. Since this loop is executing on the order of seconds, my initial reaction is that the simpler code might be preferred.

Reply:
That's true. I think it's fine in here.

Was the intention to only ping a subset in the background loop, but leave the graph unchanged, or should it also apply to the graph (UI) ?

I know that your primary concern was the number of time series in prometheus, but I'm thinking that it might be confusing to have different behaviors in different places.

Reply:
I meant to follow up on this :) yeah i think it should also be in the UI.

Reply:
OK, sounds great. Then I'll hold off with the review until then.

Happy Monday btw!